test-source/test/ceylon/time/testDuration.ceylon

```diff
@@ -14,3 +14,24 @@ import ceylon.time {
 parameters (`value scalableDurationTests`)
 shared test void testScalableDuration(Duration expectedDuration, Integer scale)
         => assertEquals { expected = expectedDuration; actual = scale ** Duration(1000); };
+
+shared test void testPlus()
+        => assertEquals (Duration.zero, Duration(-4567) + Duration(4567));
+
+shared test void testInv()
+        => assertEquals (Duration(-99), -Duration(99));
+
+shared test void testInvZero()
+        => assertEquals (Duration.zero, -Duration.zero);
+
+shared test void testMinus()
+        => assertEquals (Duration.zero, Duration(4567) - Duration(4567));
+        
+shared test void testCompare()
+        => assertEquals (Comparison.smaller, Duration(345) <=> Duration(4567));
+            
+shared test void testCompare()
+        => assertEquals (Comparison.greater, Duration(3450) <=> Duration(456));
+                
+shared test void testCompare()
+        => assertEquals (Comparison.equal, Duration(345) <=> Duration(346));
```

Comment:
345 and 346 obviously should not be considered equal?

Reply:
Yes, of course not, at least not if it's milliseconds of a duration. :smiley: 
Sorry—fixed.